pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -441,17 +434,93 @@ func setNodeScoresToZero(nodes []v1.Node) (priorityList schedulerapi.HostPriorit
 	return
 }
 
+func getRequestedVolumeSet(ns string, requestedVolumes []v1.Volume) (volumeSet map[string]struct{}, volumeCount int) {
+	// create a set of all volumes needed
+	volumeSet = map[string]struct{}{}
+	for _, volume := range requestedVolumes {
+		// not the same as len(pvNamesForRequestedVolumes). PV-s for a new pod might not be cached yet.
+		volumeCount++
+		if volume.PersistentVolumeClaim != nil {
+			if v, ok := pvcToPvMapCache.Load(getQualifiedName(ns, volume.PersistentVolumeClaim.ClaimName)); ok {
+				entry := v.(*pvcToPVEntry)
+				if !strings.EqualFold(entry.AccessMode, string(v1.ReadWriteMany)) &&
+					!strings.EqualFold(entry.AccessMode, string(v1.ReadOnlyMany)) {
+					if v, ok := pvToVolumeMapCache.Load(entry.VolumeName); ok {
+						diskName := v.(string)
+						volumeSet[diskName] = struct{}{}
```

Comment:
np: Because we already do some many lookups for volumes, pvs and pvcs here, we should name this map to signify what it actually is to avoid any more confusion to the reader. Should be something like `diskRequestedByPod`.